### DISCUSSION TITLE
Pretrain some sections of a model, then initialize those parts when training a full model

### DISCUSSION THREAD

**Discussion by stanleyshly** [Lightning-AI/pytorch-lightning#6473] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I need to pretrain the encode and decode section of an autoencoder first, then later attach a transformer in the middle of the encode and decode section. When I load the weights of the encode and decode section when pretraining it first, while initializing the weights of the transformer section, will I get an error about missing layers?

**Comment by carmocca** [comment, marked as answer]
You can use the `strict` flag of `load_from_checkpoint` to avoid the missing layer failure:

https://github.com/PyTorchLightning/pytorch-lightning/blob/079fe9bc0908fffdd55a08f7321a199bceaef6f8/pytorch_lightning/core/saving.py#L94-L95
